docs/regression_test_api.rst

```diff
@@ -204,7 +204,7 @@ You can attach arbitrary functions to run before or after any pipeline stage, wh
 Multiple hooks can be attached before or after the same pipeline stage, in which case the order of execution will match the order in which the functions are defined in the class body of the test.
 A single hook can also be applied to multiple stages and it will be executed multiple times.
 All pipeline hooks of a test class are inherited by its subclasses.
-Subclasses may override a pipeline hook of their parents by redefining the hook function and re-attaching it at the same pipeline stage.
+Subclasses may override a pipeline hook of their parents by redefining the hook function, where the overriding function may be reattached to any pipeline stage.
```

Comment:
What about something like this?

```suggestion
Subclasses may override a pipeline hook of their parents by redefining the hook function. The overriding function will not be a pipeline hook anymore, unless explicitly decorated, in which case it can be attached to any pipeline stage.
```

Reply:
What about this? _The overriding function will not be a pipeline hook unless explicitly decorated, and it may be reattached to any pipeline stage._

Reply:
Sorry, I just found the time to reply to this. Yes, your version reads better.